slack/web/classes/blocks.py

```diff
@@ -104,105 +179,171 @@ def __init__(
         *,
         image_url: str,
         alt_text: str,
-        title: Optional[str] = None,
+        title: Optional[Union[str, dict, TextObject]] = None,
         block_id: Optional[str] = None,
+        **others: dict,
     ):
+        """A simple image block, designed to make those cat photos really pop.
+        https://api.slack.com/reference/block-kit/blocks#image
         """
-        A simple image block, designed to make those cat photos really pop.
-
-        https://api.slack.com/reference/messaging/blocks#image
-
-        Args:
```

Comment:
The cost to maintain this part is not small. In this PR, I've added valid type hints to all the arguments in constructor. We can have args in docstring when it's really necessary but for most cases, let me replace them with type hints and encourage developers to access the latest API docs.

```
Help on class SectionBlock in module slack.web.classes.blocks:

class SectionBlock(Block)
 |  SectionBlock(*, block_id: Union[str, NoneType] = None, text: Union[str, dict, slack.web.classes.objects.TextObject] = None, fields: List[Union[str, dict, slack.web.classes.objects.TextObject]] = None, accessory: Union[dict, slack.web.classes.elements.BlockElement, NoneType] = None, **others: dict)
 |  
 |  Blocks are a series of components that can be combined
 |  to create visually rich and compellingly interactive messages.
 |  https://api.slack.com/reference/block-kit/blocks
 |  
 |  Method resolution order:
 |      SectionBlock
 |      Block
 |      slack.web.classes.JsonObject
 |      slack.web.classes.BaseObject
 |      builtins.object
 |  
 |  Methods defined here:
 |  
 |  __init__(self, *, block_id: Union[str, NoneType] = None, text: Union[str, dict, slack.web.classes.objects.TextObject] = None, fields: List[Union[str, dict, slack.web.classes.objects.TextObject]] = None, accessory: Union[dict, slack.web.classes.elements.BlockElement, NoneType] = None, **others: dict)
 |      A section is one of the most flexible blocks available.
 |      https://api.slack.com/reference/block-kit/blocks#section
 |  
 |  validate_fields_length(self)
 |  
 |  validate_text_or_fields_populated(self)
 |  
 |  ----------------------------------------------------------------------
 |  Readonly properties defined here:
 |  
 |  attributes
 |      set() -> new empty set object
 |      set(iterable) -> new set object
 |      
 |      Build an unordered collection of unique elements.
```